js/fieldmanager-gallery.js

```diff
@@ -1,229 +1,207 @@
-	parent.find( '.fm-gallery-id' ).val('');
-	parent.find( '.gallery-wrapper' ).html( '' );
-	fm_gallery_frame[ parent.find( '.fm-gallery-button' ).attr('id') ] = false;
-});
-
-$( document ).on( 'click', '.gallery-wrapper a', function( event ){
-	event.preventDefault();
-	$(this).closest('.gallery-wrapper').siblings('.fm-gallery-button').click();
-} );
-
-$( document ).on( 'click', '.fm-gallery-button', function( event ) {
-	var $el = $(this);
-	event.preventDefault();
-
-	// If the gallery frame already exists, reopen it.
-	if ( fm_gallery_frame[ $el.attr('id') ] ) {
-		fm_gallery_frame[ $el.attr('id') ].open();
-		return;
-	}
-	var selectedImages = [],
-		inputVal = $el.parent().find('.fm-gallery-id').val();
-
-	if ( inputVal.length ) {
-		selectedImages = inputVal.split(',');
-	}
-
-	// Modification to gallery editing workflow
-	var galleryFrame = wp.media.view.MediaFrame.Post.extend({
-
-		createStates: function() {
-			var options = this.options;
-
-			this.states.add([
-
-				new wp.media.controller.Library({
-					id:         'gallery',
-					title:      options.title,
-					priority:   40,
-					toolbar:    'main-gallery',
-					filterable: 'uploaded',
-					multiple:   'add',
-					editable:   false,
-
-					library:  wp.media.query( _.defaults({
-						type: 'image'
-					}, options.library ) )
-				}),
-
-				new wp.media.controller.GalleryEdit({
-					library: options.selection,
-					editing: options.editing,
-					menu:    'gallery'
-				}),
-
-				new wp.media.controller.GalleryAdd({
-				})
-
-			]);
-
+	/**
+	 * Clicking the gallery button.
+	 *
+	 * This will open a gallery frame to edit or create a gallery.
+	 */
+	$document.on( 'click', '.fm-gallery-button', function( event ) {
+		var $el = $( this );
+		event.preventDefault();
+
+		// If the gallery frame already exists, reopen it.
+		if ( fm_gallery_frame[ $el.attr( 'id' ) ] ) {
+			fm_gallery_frame[ $el.attr( 'id' ) ].open();
+			return;
 		}
 
-	});
+		var selectedImages = [];
+		var inputVal = $el.parent().find( '.fm-gallery-id' ).val();
 
-	var query_args = {
-		'type': 'image',
-		'post__in': selectedImages,
-		'orderby': 'post__in',
-		'perPage': -1
-	};
-
-	var attachments = wp.media.query( query_args );
-	var selection = new wp.media.model.Selection( attachments.models, {
-		props:    attachments.props.toJSON(),
-		multiple: true
-	});
+		if ( inputVal.length ) {
+			selectedImages = inputVal.split( ',' );
+		}
 
-	selection.more().done( function() {
-		// Break ties with the query.
-		selection.props.set({ query: false });
-		selection.unmirror();
-		selection.props.unset('orderby');
-	});
+		/**
+		 * Modification to gallery editing workflow
+		 *
+		 * We also need to check for the `post_frame` variable. This is to circumvent
+		 * the Media Explorer overlay onto the `wp.media.view.MediaFrame.Post` object.
+		 */
+		var media_frame_object = ( 'function' === typeof post_frame ) ? post_frame : wp.media.view.MediaFrame.Post;
```

Comment:
The [JS coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#type-checks) advise not using Yoda type checks.